pytest/tests/mocknet/load_test_spoon.py

```diff
@@ -110,192 +110,288 @@ def check_slow_blocks(initial_metrics, final_metrics):
+
+    def __parse_args(self):
+        parser = argparse.ArgumentParser(description='Run a load test')
+        parser.add_argument('--chain-id', required=True)
+        parser.add_argument('--pattern', required=False)
+        parser.add_argument('--epoch-length', type=int, required=True)
+        parser.add_argument('--num-nodes', type=int, required=True)
+        parser.add_argument('--max-tps', type=float, required=True)
+        parser.add_argument('--increasing-stakes', type=float, default=0.0)
+        parser.add_argument(
+            '--progressive-upgrade',
+            default=False,
+            action='store_true',
+        )
+        parser.add_argument('--skip-load', default=False, action='store_true')
+        parser.add_argument('--skip-setup', default=False, action='store_true')
+        parser.add_argument(
+            '--skip-restart',
+            default=False,
+            action='store_true',
+        )
+        parser.add_argument('--no-sharding', default=False, action='store_true')
+        parser.add_argument('--num-seats', type=int, required=True)
+
+        parser.add_argument('--test-timeout', type=int, default=12 * 60 * 60)
+        parser.add_argument(
+            '--contract-deploy-time',
+            type=int,
+            default=10 * mocknet.NUM_ACCOUNTS,
+            help=
+            'We need to slowly deploy contracts, otherwise we stall out the nodes',
+        )
+
+        # The flag is no longer needed but is kept for backwards-compatibility.
+        parser.add_argument('--script', required=False)
+
+        args = parser.parse_args()
+
+        self.chain_id = args.chain_id
+        self.pattern = args.pattern
+        self.epoch_length = args.epoch_length
+        self.num_nodes = args.num_nodes
+        self.max_tps = args.max_tps
+
+        self.increasing_stakes = args.increasing_stakes
+        self.progressive_upgrade = args.progressive_upgrade
+        self.num_seats = args.num_seats
+
+        self.skip_setup = args.skip_setup
+        self.skip_restart = args.skip_restart
+        self.skip_load = args.skip_load
+        self.no_sharding = args.no_sharding
+
+        self.test_timeout = args.test_timeout
+        self.contract_deploy_time = args.contract_deploy_time
+
+        assert self.epoch_length > 0
+        assert self.num_nodes > 0
+        assert self.max_tps > 0
+
+    def __prepare_nodes(self):
+        all_nodes = mocknet.get_nodes(pattern=self.pattern)
+        random.shuffle(all_nodes)
+        assert len(all_nodes) > self.num_nodes, \
+            f'Need at least one RPC node. ' \
+            f'Nodes available in mocknet: {len(all_nodes)}' \
+            f'Requested validator nodes num: {self.num_nodes}'
+
+        self.all_nodes = all_nodes
+        self.validator_nodes = all_nodes[:self.num_nodes]
+        self.rpc_nodes = all_nodes[self.num_nodes:]
+        self.archival_node = self.rpc_nodes[0]
+
+        logger.info(f'validator_nodes: {self.validator_nodes}')
```

Comment:
I think this would be more useful:

```diff
diff --git a/pytest/tests/mocknet/load_test_spoon.py b/pytest/tests/mocknet/load_test_spoon.py
index 6561d837f..cf1f30491 100755
--- a/pytest/tests/mocknet/load_test_spoon.py
+++ b/pytest/tests/mocknet/load_test_spoon.py
@@ -222,8 +222,8 @@ class LoadTestSpoon:
         self.rpc_nodes = all_nodes[self.num_nodes:]
         self.archival_node = self.rpc_nodes[0]
 
-        logger.info(f'validator_nodes: {self.validator_nodes}')
-        logger.info(f'rpc_nodes      : {self.rpc_nodes}')
+        logger.info(f'validator_nodes: {[n.instance_name for n in self.validator_nodes]}')
+        logger.info(f'rpc_nodes      : {[n.instance_name for n in self.rpc_nodes]}')
         logger.info(f'archival node  : {self.archival_node.instance_name}')
 
     def __prepare_upgrade_schedule(self):
```